### PR TITLE
Add Duo Universal Prompt Support to IS 5.10

### PR DESCRIPTION
**Pull request by amanda-ariyaratne**
## Purpose

The changes contained in this PR makes the Duo connector supporting the universal prompt compatible for IS 5.10.
- Downgrade carbon-identity-framework dependency is to `v5.17.5`
- Remove usages of ServiceURLBuilder because it is not supported by the above version.
- Add `duoauthenticationendpoint` webapp to display error messages.
- Add new config to set error page url.

This will remove support for the Duo Traditional Prompt which was deprecated by Duo.

## Related Issues
- https://github.com/wso2/product-is/issues/14951

## Related PRs
- https://github.com/wso2-extensions/identity-outbound-auth-duo/pull/54

### CODE REVIEW

File: component/authentication-endpoint/pom.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (ashensw) [inline component/authentication-endpoint/pom.xml]: Shall we update these license headers? Check all the new files.

File: component/authentication-endpoint/pom.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    <packaging>war</packaging>
+    <name>WSO2 Carbon - Identity Application Authentication Duo Endpoint</name>
+    <description>Identity Application Authentication Duo Endpoint</description>
+    <url>http://wso2.org</url>
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>1.7</source>
```
Comment (ashensw) [inline component/authentication-endpoint/pom.xml]: Let's use 1.8

File: component/authentication-endpoint/pom.xml (hunk 1 above)
Comment (amanda-ariyaratne) [reply component/authentication-endpoint/pom.xml]: Fixed with 208f4a5beabb544f608585d16512c1b9e45a2117

File: component/authentication-endpoint/pom.xml (hunk 2 above)
Comment (amanda-ariyaratne) [reply component/authentication-endpoint/pom.xml]: Fixed with 37d2b2c8f6ad24532b4ffc30f2f05e870f90351b

**Review by ashensw — DISMISSED** [review]

**Review by ashensw — APPROVED** [review]

**Pull request merged**